src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.Id.MachineName.cs

```diff
@@ -3,11 +3,19 @@
     public partial class Activity
     {
         #region private
-        private string generateUniquePrefix()
+
+        static Activity()
```

Comment:
Having static class constructors like this place create an inefficiency (there has to be a check before any method in this class.     Since it is so easy to avoid this (because we can do the check in really just one place), that is better.